client/src/App.js

```diff
@@ -40,43 +44,69 @@ const App = () => {
   });
   const [showToast, toggleToast] = useState(false);
   const [userAuthenticated, setAuthenticated] = useState(false);
+  const [userDetails, setUserDetails] = useState({
+    id: null,
+    name: ''
+  });
 
   const activateToast = (text, variant = 'neutral', button = 'CLOSE') => {
     setToastProperties({ text, variant, button });
     toggleToast(true);
   };
 
+  const contextProps = {
+    activateToast,
+    userAuthenticated,
+    setAuthenticated,
+    userDetails,
+    setUserDetails
+  };
+
   useEffect(() => {
-    setAuthenticated(Boolean(localStorage.getItem('jwtToken')));
+    const jwtToken = localStorage.getItem('jwtToken');
+
+    if (jwtToken) {
+      try {
+        const data = jwtVerify(jwtToken, process.env.REACT_APP_JWT_SECRET).payload;
+        setUserDetails({ name: data.name, id: data._id });
+        setAuthenticated(true);
+      } catch (error) {
+        if (error.name === 'TokenExpiredError') localStorage.removeItem('jwtToken');
+        activateToast('Your session has expired, please log in again.', 'error');
+      }
+    }
   }, []);
 
   return (
     <MuiThemeProvider theme={theme}>
       <BrowserRouter>
-        {/* Placeholder user object */}
-        <NavBar user={{ name: 'Joe' }} authenticated={userAuthenticated} setAuthenticated={setAuthenticated}/>
+        <NavBar
+          userDetails={userDetails}
+          userAuthenticated={userAuthenticated}
+          setAuthenticated={setAuthenticated}
+        />
 
         {/* Routes */}
         {/*- Base route uses a Redirect Component to redirect to
             /signup. Change render to component with the home page
             component if changing landing page.
+
+          - wrap export in withPageContext to retrieve global state like:
+          - - Authentication
+          - - User details
+          - Other global variables can be put in the variable contextProps to pass it
+          - to each page.
         */}
-        <ToastContext.Provider value={activateToast}>
+        <PageContext.Provider value={contextProps}>
           <Route
             exact
             path="/"
-            render={() => <Redirect to={userAuthenticated ? '/profile' : '/signup'}/>}
-          />
-          <Route
-            path="/signup"
-            render={(routerProps) => <SignUp setAuthenticated={setAuthenticated} {...routerProps} />}
-          />
-          <Route
-            path="/login"
-            render={(routerProps) => <Login setAuthenticated={setAuthenticated} {...routerProps} />}
+            render={() => <Redirect to={userAuthenticated ? '/profile' : '/signup'} />}
           />
-          <Route path="/profiles/:id" component={ProfilePage}/>
-        </ToastContext.Provider>
+          <Route path="/signup" component={SignUp} />
+          <Route path="/login" component={Login} />
+          <Route path="/profile/:id?" component={ProfilePage} />
```

Comment:
Related to below regarding the plural resource in the path. Maybe I'm not understanding, but generally a restful get req would be "profiles/id..."

Reply:
@tsyaeger I've never seen a frontend /profiles url, but I may be wrong about the right way.
Also if there is no id in url it goes to user's profile so i just thought it sounded better, but I'm open to discussion of course

Reply:
Yeah -- let's chat. I never thought about front end paths being different. And we agree that the id should definitely remain.